### PR TITLE
add python bindings to this repo

### PR DESCRIPTION
**Pull request by dpaiton**
Adds the `hyperdrivepy` bindings to this repo so they can be centrally located
Modifies `Cargo.toml` files to all share the same version as the `hyperdrive-rs` workspace file

All `*.rs` changes were moving stuff around, lint, etc. The workflow changes were substantive.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,10 +7,11 @@ members = [
     "crates/hyperdrive-math",
     "crates/hyperdrive-wrappers",
     "crates/test-utils",
+    "bindings/hyperdrivepy",
 ]
 
-[build]
-build = "build.rs"
+# [build]
+# build = "build.rs"
```
Comment (sentilesdal) [inline Cargo.toml]: is this supposed to be commented?  get rid of it instead?

File: bindings/hyperdrivepy/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# hyperdrivepy
+
```
Comment (sentilesdal) [inline bindings/hyperdrivepy/README.md:2]: Rust to Python bindings?

File: bindings/hyperdrivepy/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[project]
+name = "hyperdrivepy"
+version = "0.14.0"
+requires-python = ">=3.7"
```
Comment (sentilesdal) [inline bindings/hyperdrivepy/pyproject.toml:4]: 3.7?  can we bump? do we care?

File: bindings/hyperdrivepy/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+classifiers = [
+  "Programming Language :: Rust",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+readme = "README.md"
+
+[project.urls]
+Homepage = "https://github.com/delvtech/hyperdrive-bindings"
```
Comment (sentilesdal) [inline bindings/hyperdrivepy/pyproject.toml]: this is ded. update?

File: bindings/hyperdrivepy/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+Issues = "https://github.com/delvtech/hyperdrive-bindings/issues"
```
Comment (sentilesdal) [inline bindings/hyperdrivepy/pyproject.toml]: same

File: bindings/hyperdrivepy/python/hyperdrivepy/pypechain_types/__init__.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+"""Export all types from generated files.
+
+DO NOT EDIT.  This file was generated by pypechain.  See documentation at
+https://github.com/delvtech/pypechain"""
+
+# python 3.10 causes this warning when sibling/child files import from one of the files listed here.
+# remove this pylint diable when we upgreade to 3.11
```
Comment (sentilesdal) [inline bindings/hyperdrivepy/python/hyperdrivepy/pypechain_types/__init__.py:7]: *upgrade

**Review by sentilesdal — APPROVED** [review]
I assume mostly copy pasta so i skimmed the whole rust files.  looks good overall, nice!

File: bindings/hyperdrivepy/pyproject.toml (hunk 3 above)
Comment (dpaiton) [reply bindings/hyperdrivepy/pyproject.toml:4]: we don't care too much; I think it was pyo3 that set this limit. Once we bump `agent0` to 3.11 we can bump this one too and remove all `from __future__` imports.

**Pull request merged**